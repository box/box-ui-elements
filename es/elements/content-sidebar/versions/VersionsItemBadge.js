function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
/**
 * 
 * @file Versions Item Badge component
 * @author Box
 */
import * as React from 'react';
import { FormattedMessage, injectIntl } from 'react-intl';
import messages from './messages';
import './VersionsItemBadge.scss';
const VersionsItemBadge = ({
  intl,
  versionNumber
}) => {
  const intlValues = {
    versionNumber
  };
  return /*#__PURE__*/React.createElement("div", {
    "aria-label": intl.formatMessage(messages.versionNumberLabel, intlValues),
    className: "bcs-VersionsItemBadge"
  }, /*#__PURE__*/React.createElement(FormattedMessage, _extends({}, messages.versionNumberBadge, {
    values: intlValues
  })));
};
export default injectIntl(VersionsItemBadge);
//# sourceMappingURL=VersionsItemBadge.js.map