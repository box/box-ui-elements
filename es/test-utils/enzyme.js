function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
/* eslint-disable import/no-extraneous-dependencies */
import * as React from 'react';
import { mount as baseMount, shallow as baseShallow } from 'enzyme';

// Global providers that need setup
import { ThemeProvider } from 'styled-components';
import defaultTheme from '../styles/theme';

// Wrap this around every component so they can use app-wide context.
// Prevents breaking due to missing providers.
// See https://github.com/airbnb/enzyme/blob/master/docs/api/ReactWrapper/getWrappingComponent.md
/* eslint-disable react/prop-types */
const Wrappers = ({
  children
}) => {
  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: defaultTheme
  }, children);
};

/**
 * mount() from Enzyme but with required app-wide context providers included by default
 */
const mountConnected = (element, options = {}) => baseMount(element, _objectSpread({
  wrappingComponent: Wrappers
}, options));

/**
 * shallow() from Enzyme but with required app-wide context providers included by default.
 */
const shallowConnected = (element, options = {}) => baseShallow(element, _objectSpread({
  wrappingComponent: Wrappers
}, options));

// Re-export everything from Enzyme
export * from 'enzyme';
// Export wrapped methods
export { mountConnected, shallowConnected };
//# sourceMappingURL=enzyme.js.map