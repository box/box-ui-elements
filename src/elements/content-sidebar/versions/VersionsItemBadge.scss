@import '../../common/variables';

.bcs-VersionsItemBadge {
    padding: 10px 8px;
    color: $bdl-box-blue;
    font-size: 18px;
    text-align: center;
    background-color: $selected-blue-background;
    border-radius: $bdl-border-radius-size;
}
