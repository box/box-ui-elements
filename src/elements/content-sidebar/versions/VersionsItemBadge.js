/**
 * @flow
 * @file Versions Item Badge component
 * @author Box
 */
import * as React from 'react';
import { FormattedMessage, injectIntl } from 'react-intl';
import type { intlShape } from 'react-intl';
import messages from './messages';
import './VersionsItemBadge.scss';

type Props = {
    intl: intlShape,
    versionNumber: string,
};

const VersionsItemBadge = ({ intl, versionNumber }: Props) => {
    const intlValues = { versionNumber };

    return (
        <div aria-label={intl.formatMessage(messages.versionNumberLabel, intlValues)} className="bcs-VersionsItemBadge">
            <FormattedMessage {...messages.versionNumberBadge} values={intlValues} />
        </div>
    );
};

export default injectIntl(VersionsItemBadge);
