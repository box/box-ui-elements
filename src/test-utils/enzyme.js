/* eslint-disable import/no-extraneous-dependencies */
import * as React from 'react';
import { mount as baseMount, shallow as baseShallow } from 'enzyme';

// Global providers that need setup
import { ThemeProvider } from 'styled-components';
import defaultTheme from '../styles/theme';

// Wrap this around every component so they can use app-wide context.
// Prevents breaking due to missing providers.
// See https://github.com/airbnb/enzyme/blob/master/docs/api/ReactWrapper/getWrappingComponent.md
/* eslint-disable react/prop-types */
const Wrappers = ({ children }) => {
    return <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>;
};

/**
 * mount() from Enzyme but with required app-wide context providers included by default
 */
const mountConnected = (element, options = {}) => baseMount(element, { wrappingComponent: Wrappers, ...options });

/**
 * shallow() from Enzyme but with required app-wide context providers included by default.
 */
const shallowConnected = (element, options = {}) => baseShallow(element, { wrappingComponent: Wrappers, ...options });

// Re-export everything from Enzyme
export * from 'enzyme';
// Export wrapped methods
export { mountConnected, shallowConnected };
