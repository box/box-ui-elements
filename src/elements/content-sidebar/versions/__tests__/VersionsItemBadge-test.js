import * as React from 'react';
import { render } from 'enzyme/build';
import VersionsItemBadge from '../VersionsItemBadge';

describe('elements/content-sidebar/versions/VersionsItemBadge', () => {
    const getWrapper = (props = {}) => render(<VersionsItemBadge {...props} />);

    describe('render', () => {
        test('should match its snapshot', () => {
            const wrapper = getWrapper({ versionNumber: '1' });
            expect(wrapper).toMatchSnapshot();
        });
    });
});
