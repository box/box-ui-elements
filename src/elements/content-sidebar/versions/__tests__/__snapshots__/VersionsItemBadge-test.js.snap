// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`elements/content-sidebar/versions/VersionsItemBadge render should match its snapshot 1`] = `
<div
  aria-label="Version number 1"
  class="bcs-VersionsItemBadge"
>
  <div />
</div>
`;
